File: containerwithmostwater.cpp
class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int maxar = 0;
        int left = 0;
        int right = height.size() - 1;
        // Took two variables"left" and "right" with great difference so as to
        // consider all areas.
        while (left < right)
        {

            int area = 0;
            // AS area= length*breadth, therefore
            int length = min(height[left], height[right]);
            int diff = right - left;
            int breadth = diff;
            area = length * breadth;
            maxar = max(maxar, area);

            if (height[left] < height[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }

        return maxar;
    }
};

File: containerwithmostwater.java
class Solution {
    public int maxArea(int[] height) {
        int ar = 0;
        int start = 0;
        int end = height.length - 1;
        int l = 0;
        int b = 0;
        while (start < end) {
            b = end - start;
            if (height[start] > height[end]) {
                l = height[end];
                end--;
            } else if (height[end] > height[start]) {
                l = height[start];
                start++;
            } else {
                l = height[start];
                start++;
                end--;
            }
            ar = Math.max(ar, l * b);
        }
        return ar;
    }
}